client/state/initial-state.js

```diff
@@ -88,11 +90,23 @@ export function persistOnChange( reduxStore, serializeState = serialize ) {
 		global.window.addEventListener( 'beforeunload', throttledSaveState.flush );
 	}
 
-	reduxStore.subscribe( throttledSaveState );
+	unsubscribe = reduxStore.subscribe( throttledSaveState );
 
 	return reduxStore;
 }
 
+/**
+ * Removes redux state change listener, cancels any pending writes,
+ * and clears all persisted state.
+ *
+ * @param {function} onClear function to call on clearing of persisted state
+ */
+export function shutdown( onClear ) {
+	unsubscribe();
```

Comment:
I'm not sure this would work with our throttle usage in `initial-state`. How about checking for if the user is logged out and then returning early in throttledSaveState?